qt/scientific_interfaces/Indirect/IndirectFitDataTableModel.cpp

```diff
@@ -0,0 +1,349 @@
+      m_resolutionsSingle(std::make_unique<std::vector<std::weak_ptr<Mantid::API::MatrixWorkspace>>>()),
+      m_fittingDataMultiple(std::make_unique<std::vector<IndirectFitData>>()),
+      m_resolutionsMultiple(std::make_unique<std::vector<std::weak_ptr<Mantid::API::MatrixWorkspace>>>()) {
+
+  m_fittingData = m_fittingDataSingle.get();
+  m_resolutions = m_resolutionsSingle.get();
+}
+
+bool IndirectFitDataTableModel::hasWorkspace(std::string const &workspaceName) const {
+  auto const names = getWorkspaceNames();
+  auto const iter = std::find(names.begin(), names.end(), workspaceName);
+  return iter != names.end();
+}
+
+Mantid::API::MatrixWorkspace_sptr IndirectFitDataTableModel::getWorkspace(TableDatasetIndex index) const {
+  if (index < m_fittingData->size())
+    return m_fittingData->at(index.value).workspace();
+  return nullptr;
+}
+
+FunctionModelSpectra IndirectFitDataTableModel::getSpectra(TableDatasetIndex index) const {
+  if (index < m_fittingData->size())
+    return m_fittingData->at(index.value).spectra();
+  return FunctionModelSpectra("");
+}
+
+TableDatasetIndex IndirectFitDataTableModel::getNumberOfWorkspaces() const {
+  return TableDatasetIndex{m_fittingData->size()};
+}
+
+bool IndirectFitDataTableModel::isMultiFit() const { return getNumberOfWorkspaces().value > 1; }
+
+size_t IndirectFitDataTableModel::getNumberOfSpectra(TableDatasetIndex index) const {
+  if (index < m_fittingData->size())
+    return m_fittingData->at(index.value).numberOfSpectra().value;
+  else
+    throw std::runtime_error("Cannot find the number of spectra for a workspace: the workspace "
+                             "index provided is too large.");
+  return 1;
+}
+
+size_t IndirectFitDataTableModel::getNumberOfDomains() const {
+  size_t sum{0};
+  for (auto fittingData : *m_fittingData) {
+    sum += fittingData.numberOfSpectra().value;
+  }
+  return sum;
+}
+
+std::vector<double> IndirectFitDataTableModel::getQValuesForData() const {
+  std::vector<double> qValues;
+  for (auto &fitData : *m_fittingData) {
+    auto indexQValues = fitData.getQValues();
+    qValues.insert(std::end(qValues), std::begin(indexQValues), std::end(indexQValues));
+  }
+  return qValues;
+}
+
+std::vector<std::pair<std::string, size_t>> IndirectFitDataTableModel::getResolutionsForFit() const {
+  std::vector<std::pair<std::string, size_t>> resolutionVector;
+  for (size_t index = 0; index < m_resolutions->size(); ++index) {
+    auto resolutionWorkspace = m_resolutions->at(index).lock();
+    auto spectra = getSpectra(TableDatasetIndex{index});
+    if (!resolutionWorkspace) {
+      for (auto &spectraIndex : spectra) {
+        resolutionVector.emplace_back("", spectraIndex.value);
+      }
+      continue;
+    }
+
+    auto singleSpectraResolution = m_resolutions->at(index).lock()->getNumberHistograms() == 1;
+    for (auto &spectraIndex : spectra) {
+      auto resolutionIndex = singleSpectraResolution ? 0 : spectraIndex.value;
```

Comment:
We can do `auto const` for a lot of the variables in this function, and the function above